cmd/cfgenerator/main.go

```diff
@@ -116,11 +136,25 @@ func run(interpreterName string, inputPath string, outputPath string, volumes []
 	}
 	defer input.Close()
 
-	output, err := file.OpenOutput(outputPath)
+	content, err := internal.Generate(runtime, input, volumes)
 	if err != nil {
-		return fmt.Errorf("can't open output file '%s': %v", outputPath, err)
+		return fmt.Errorf("can't generate content: %v", err)
+	}
+
+	outputs := make([]*os.File, len(outputPaths))
+	for i, outputPath := range outputPaths {
+		output, err := file.OpenOutput(outputPath)
+		if err != nil {
+			return fmt.Errorf("can't open output file '%s': %v", outputPath, err)
+		}
+		defer output.Close()
+
+		outputs[i] = output
+	}
+
+	for i := range outputs {
```

Comment:
I have done this part outside of the previous loop to ensure all the file paths given could be open and have less chances to end up with an inconsistent state where only some files would be generated but not all.